diff --git a/YusUtilityObj.m b/YusUtilityObj.m
index 4c55788..5cd2e5d 100644
--- a/YusUtilityObj.m
+++ b/YusUtilityObj.m
@@ -12,6 +12,12 @@ classdef YusUtilityObj < handle
         
         % Stores throughputs
         Throughput
+        
+        % Metrics step size
+        MetricsStepSize
+        
+        % Indices of cells of interest
+        cellOfInterestIdx
     end
     methods
         function obj = YusUtilityObj(param, numSlotsSim)
@@ -27,7 +33,9 @@ classdef YusUtilityObj < handle
             % stores UE indices and extra two item for cell throughput and
             % theoretical peak throughput. The third index stores site
             % indices. The fourth index stores slot numbers.
-            obj.Throughput = zeros(2, d3+2, d2, d1);
+            obj.MetricsStepSize = param.MetricsStepSize;
+            d4 = param.NumMetricsSteps;
+            obj.Throughput = zeros(2, d3+2, d2, d4);
         end
         
         function pushSlotNum(obj,slotNum)
@@ -76,7 +84,8 @@ classdef YusUtilityObj < handle
         
         function storeThroughput(obj,throughputServed, siteIdx, slotNum)
             % Stores throughput in obj.Throughput
-            obj.Throughput(:,:,siteIdx,slotNum) = throughputServed;
+            d4Idx = slotNum/obj.MetricsStepSize;
+            obj.Throughput(:,:,siteIdx,d4Idx) = throughputServed;
         end
         
         function SaveFile(obj)
diff --git a/plotThroughputCDF.m b/plotThroughputCDF.m
index ae0b1b4..482b4e4 100644
--- a/plotThroughputCDF.m
+++ b/plotThroughputCDF.m
@@ -1,17 +1,16 @@
 function plotThroughputCDF(YUO,varargin)
     % Plot CDF of throughputs
+    CellIntIdx = YUO.cellOfInterestIdx;
+    DLThpt = YUO.Throughput(1,1:(end-2),CellIntIdx,:);
+    ULThpt = YUO.Throughput(2,1:(end-2),CellIntIdx,:);
     if isempty(varargin)
-        DLThpt = YUO.Throughput(1,:,:,:);
-        ULThpt = YUO.Throughput(2,:,:,:);
         plotCDF(DLThpt(:),'DL');
         plotCDF(ULThpt(:),'UL');
     else
         switch varargin{1}
             case 'DL'
-                DLThpt = YUO.Throughput(1,:,:,:);
                 plotCDF(DLThpt(:),'DL');
             case 'UL'
-                ULThpt = YUO.Throughput(2,:,:,:);
                 plotCDF(ULThpt(:),'UL');
         end
     end
